### ISSUE TITLE
Unexpected result on cortex-m7 target with float model from HelloWorld example

### ISSUE THREAD

**Issue by Evanok** [tensorflow/tflite-micro#2454]
Hello,

I am probably the issue here but I cannot make work the hello_world example on a target.
I am able to run the code on the target but the output from the model is not the expected result.
The hello_world example is working fine on my host using python.

My current process:

- clone tflite-micro on main branch
using commit: 0ec823ed2f46a183ff202725b83c8ee3e13215b2
- compile tflite micro library for cortex m7
`make -f tensorflow/lite/micro/tools/make/Makefile TARGET=cortex_m_generic TARGET_ARCH=cortex-m7+fp OPTIMIZED_KERNEL_DIR=cmsis_nn microlite
`- Link my cortex m7 firmware with tflite-micro library 
location:
` tflite-micro/gen/cortex_m_generic_cortex-m7+fp_default/lib/libtensorflow-microlite.a
`- Trying to compile the firmware and fix all the error by adding the include directory path into my cmakefile
After iterating all the include directory added:
```
-> tflite-micro
-> tflite-micro/gen/cortex_m_generic_cortex-m7+fp_default/genfiles
-> tflite-micro/tensorflow/lite/micro/tools/make/downloads/flatbuffers/include
-> tflite-micro/tensorflow/lite/micro/tools/make/downloads/gemmlowp
```
- Integrate code to invok the model on target from hello world example into my firmware code.
I am using code from: 
`tensorflow/lite/micro/examples/hello_world/hello_world_test.cc`
For now I am only testing the float model.
- Add in the source list in my cmake firmware, the binary model code
location: 
`tflite-micro/gen/cortex_m_generic_cortex-m7+fp_default/genfiles/tensorflow/lite/micro/examples/hello_world/models/hello_world_float_model_data.cc`

Finally add some logs into the tflite code integrated into my firmware to show the result:

```
(...)
for (int i = 0; i < kNumTestValues; ++i) {
	interpreter.input(0)->data.f[0] = golden_inputs[i];
	TF_LITE_ENSURE_STATUS(interpreter.Invoke());
	float y_pred = interpreter.output(0)->data.f[0];
	PARDI_TRACE("Data input is %f\n", golden_inputs[i]);
	PARDI_TRACE("Model output is %f\n", y_pred);
	PARDI_TRACE("Expected output of sin(%f) = %f\n", golden_inputs[i], sin(golden_inputs[i]));
	//TFLITE_CHECK_LE(abs(sin(golden_inputs[i]) - y_pred), epsilon);
}
(...)
```

The result on my serial console:

```
Data input is 0.000000
Model output is 0.000000
Expected output of sin(0.000000) = 0.000000
Data input is 1.000000
Model output is 1.000000
Expected output of sin(1.000000) = 0.841471
Data input is 3.000000
Model output is 3.000000
Expected output of sin(3.000000) = 0.141120
Data input is 5.000000
Model output is 5.000000
Expected output of sin(5.000000) = -0.958924
```

Basically input == output :(.

Is there something wrong in my process ?

Happy Valentine's Day!  ♥️

**Comment by Evanok** [comment]
Hello @ddavis-2015,

Thanks for the reply.

I did not add any specific CFLAGS in my cmakefile.
I just run a new test with -DTF_LITE_STATIC_MEMORY  CFLAGS. 
This is working way better. **the float model seems to work fine**. The result is not very good on the Quant int8 model.

```
float32 model
Data input is 0.000000
Model output is 0.026405
Expected output of sin(0.000000) = 0.000000
Data input is 1.000000
Model output is 0.863044
Expected output of sin(1.000000) = 0.841471
Data input is 3.000000
Model output is 0.127646
Expected output of sin(3.000000) = 0.141120
Data input is 5.000000
Model output is -0.956519
Expected output of sin(5.000000) = -0.958924
_ _ _ _ _ _ _ _ _ 
```

```
Quant int8 model
Data input is 0.770000
Model output is 0.696440
Expected output of sin(0.770000) = -0.983588
Data input is 1.570000
Model output is 0.994915
Expected output of sin(1.570000) = -0.167356
Data input is 2.300000
Model output is 0.729604
Expected output of sin(2.300000) = -0.529083
Data input is 3.140000
Model output is -0.008291
Expected output of sin(3.140000) = 0.000000
_ _ _ _ _ _ _ _ _ 
```

Did I miss a documentation ?
This repository is designed to run AI model into embedded platform but the hello world example README is only talking about way to run and test the model on host device.

I am going to try to run my own model trained model now.

**Comment by Evanok** [comment]
I am closing the issue. The result are all good now.
